spyder/plugins/maininterpreter.py

```diff
@@ -77,19 +78,25 @@ def setup_page(self):
             filters = _("Executables")+" (*.exe)"
         else:
             filters = None
-        pyexec_file = self.create_browsefile('', 'executable', filters=filters)
-        for le in self.lineedits:
-            if self.lineedits[le][0] == 'executable':
-                self.pyexec_edit = le
-        def_exec_radio.toggled.connect(pyexec_file.setDisabled)
-        self.cus_exec_radio.toggled.connect(pyexec_file.setEnabled)
+
         pyexec_layout = QVBoxLayout()
         pyexec_layout.addWidget(pyexec_label)
         pyexec_layout.addWidget(def_exec_radio)
         pyexec_layout.addWidget(self.cus_exec_radio)
-        pyexec_layout.addWidget(pyexec_file)
+        self.cus_exec_combo = self.create_file_combobox(
+                                            _('Recent custom interpreters'),
+                                            self.get_option('custom_list'),

```

Comment:
So here, before setting the custom list of interpreters, we need to validate that they still exist and remove the ones that don't.

Reply:
Maybe it would be better to do the validation in the `__init__` method?

Reply:
Sure, that's ok too.